tests/storage/mpathconf_test.py

```diff
@@ -49,6 +52,32 @@
 }
 """
 
+FAKE_MPATHD_OUTPUT = Template("""\
+defaults {
+        verbosity 2
+        polling_interval 5
+        multipath_dir "/lib64/multipath"
+        path_selector "service-time 0"
+        prio_args ""
+        user_friendly_names $defaults_value
+        find_multipaths_timeout -10
+        enable_foreign "NONE"
```

Comment:
I already commented about this - you test here only value in the middle, but we need to test:

- value is first
- value is last
- value is not first or last

This is true for all sections. So using one test config is not helpful.

For testing parsing, we don't care about actual keys - it is enough to use
only few keys, possibly always:

    section {
        key1 "value1"
        key2 "value2"
        key3 "value3"
    }

It would be nice to test sections with no key, 1 key, and multiple keys.

The parser does not care abut the actual names of the sections, it depends
only on the generic format (key, value). So we can test the parser with:

    section1 {
        key1 "value1"
        key2 "value2"
        key3 "value3"
    }
    section2 {
        key1 "value1"
    }
    section3 {
    }
    section4 {
        entry {
            key1 "value1"
            key2 "value2"
            key3 "value3"
        }
        entry {
            key1 "value1"
        }
        entry {
        }
    }

This will help to understand the parser and make the parser test very clear.
For example we can test:

    assert _parse_conf(it) == [
         Section(
             "section1", [
                 Pair("key1", "value1"),
                 ...
             ]
         ),
         ...
    ]

Reply:
I didn't ignore the point. The cases of:
- value is first
- value is last
- value is not first or last
I addressed them by putting the user_friendly_names pair in different locations in different sections.

But nevermind that. I'll separate the config example in different blocks and tests, with generic names.